OverlayPlugin.Core/MemoryProcessors/Combatant/CombatantMemory64.cs

```diff
@@ -0,0 +1,246 @@
+
+            [FieldOffset(0x30)]
+            public fixed byte Name[NameBytes];
+
+            [FieldOffset(0x74)]
+            public uint ID;
+
+            [FieldOffset(0x80)]
+            public uint BNpcID;
+
+            [FieldOffset(0x84)]
+            public uint OwnerID;
+
+            [FieldOffset(0x8C)]
+            public byte Type;
+
+            [FieldOffset(0x92)]
+            public byte EffectiveDistance;
+
+            [FieldOffset(0x94)]
+            public byte Status;
+
+            [FieldOffset(0xB0)]
+            public Single PosX;
+
+            [FieldOffset(0xB4)]
+            public Single PosY;
+
+            [FieldOffset(0xB8)]
+            public Single PosZ;
+
+            [FieldOffset(0xC0)]
+            public Single Heading;
+
+            [FieldOffset(0xD0)]
+            public Single Radius;
+
+            [FieldOffset(0x114)]
+            public int ModelStatus;
+
+            [FieldOffset(0x1C4)]
+            public int CurrentHP;
+
+            [FieldOffset(0x1C8)]
+            public int MaxHP;
+
+            [FieldOffset(0x1CC)]
+            public int CurrentMP;
+
+            [FieldOffset(0x1D0)]
+            public int MaxMP;
+
+            [FieldOffset(0x1D4)]
+            public ushort CurrentGP;
+
+            [FieldOffset(0x1D6)]
+            public ushort MaxGP;
+
+            [FieldOffset(0x1D8)]
+            public ushort CurrentCP;
+
+            [FieldOffset(0x1DA)]
+            public ushort MaxCP;
+
+            [FieldOffset(0x1DC)]
+            public short TransformationId;
+
+            [FieldOffset(0x1E2)]
+            public byte Job;
+
+            [FieldOffset(0x1E3)]
+            public byte Level;
+
+            [FieldOffset(0xCB0)]
+            public uint PCTargetID;
+
+            // TODO: this is incorrect in 6.3, please fix
+            [FieldOffset(0x19C3)]
+            public byte MonsterType;
+
+            // TODO: this is incorrect in 6.3, please fix
+            [FieldOffset(0x19DF)]
+            public byte AggressionStatus;
+
+            [FieldOffset(0x1AB8)]
+            public uint NPCTargetID;
+
+            [FieldOffset(0x1B00)]
+            public uint BNpcNameID;
+
+            [FieldOffset(0x1B1C)]
+            public ushort CurrentWorldID;
+
+            [FieldOffset(0x1B1E)]
+            public ushort WorldID;
+
+            [FieldOffset(0x1B0A)]
+            public byte WeaponId;
+
+            [FieldOffset(0x1B68)]
+            public fixed byte Effects[EffectBytes];
+
+            [FieldOffset(0x1CF0)]
+            public byte IsCasting1;
+
+            [FieldOffset(0x1CF2)]
+            public byte IsCasting2;
+
+            [FieldOffset(0x1CF4)]
+            public uint CastBuffID;
+
+            [FieldOffset(0x1CF0)]
+            public uint CastTargetID;
+
+            [FieldOffset(0x1DF4)]
+            public float CastDurationCurrent;
+
+            [FieldOffset(0x1DF8)]
+            public float CastDurationMax;
```

Comment:
These offset should also be correct based on some quick testing, but please double check.

```suggestion
            [FieldOffset(0x1B88)]
            public fixed byte Effects[EffectBytes];

            [FieldOffset(0x1D10)]
            public byte IsCasting1;

            [FieldOffset(0x1D11)]
            public byte IsCasting2;

            [FieldOffset(0x1D14)]
            public uint CastBuffID;

            [FieldOffset(0x1D20)]
            public uint CastTargetID;

            [FieldOffset(0x1D44)]
            public float CastDurationCurrent;

            [FieldOffset(0x1D48)]
            public float CastDurationMax;
```

Reply:
This looks good, tested in o12n and saw weaponID 4 for the sword+shield. Cast and effects seem fine too.